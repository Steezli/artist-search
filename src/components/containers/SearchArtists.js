import React, { PureComponent } from 'react';
import Artists from '../artists/Artists';
import { getArtists } from '../../services/MusicApi';
import Input from '../Input';


export default class ArtistList extends PureComponent {
  state = {
    artists: [],
    artist: '',
    artistCount: 0,
    offset: 0
  }


  inputChange = ({ target }) => {
    this.setState({ artist: target.value });
  }

  fetchArtists = () => {
    getArtists(this.state.artist, this.state.offset)
      .then(({ artists, count }) => { 
        this.setState({ artists, artistCount: count });
      });
  }

  onSearch = () => {
    return this.fetchArtists();
  }

  pageForward = () => {
    this.setState(state => ({ offset: state.offset + 25 }), () => {
      this.fetchArtists();

    });
  }

  pageBack = () => {
    this.setState(state => ({ offset: state.offset - 25 }), () => {
      this.fetchArtists();
    });
  }

  render() {
    return (
      <>
        <Input artist={this.state.artist}
          inputChange={this.inputChange}
          onSearch={this.onSearch}
        />
        <button onClick={() => this.pageBack()}>←</button>
        <button onClick={() => this.pageForward()}>→</button>
        <Artists artists={this.state.artists} />
      </>
    );
  }
}
