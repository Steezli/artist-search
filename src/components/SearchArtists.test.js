import React from 'react';
import { shallow } from 'enzyme';
import SearchArtists from './containers/SearchArtists';

describe('SearchArtists component', () => {
  it('renders SearchArtists', () => {
    const wrapper = shallow(<SearchArtists />);
    expect(wrapper).toMatchSnapshot();
  });
});
