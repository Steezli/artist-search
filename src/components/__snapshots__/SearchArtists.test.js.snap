// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchArtists component renders SearchArtists 1`] = `
<Fragment>
  <Input
    artist=""
    inputChange={[Function]}
    onSearch={[Function]}
  />
  <button
    onClick={[Function]}
  >
    ←
  </button>
  <button
    onClick={[Function]}
  >
    →
  </button>
  <Artists
    artists={Array []}
  />
</Fragment>
`;
